Mathlib/GroupTheory/FixedPointFree.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2024 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import Mathlib.Algebra.Group.Aut
+import Mathlib.Algebra.GroupPower.IterateHom
+import Mathlib.Data.Fintype.Card
+
+/-!
+# Fixed-point-free automorphisms
+
+This file defines fixed-point-free automorphisms and proves some basic properties.
+
+An automorphism `φ` of a group `G` is fixed-point-free if `1 : G` is the only fixed point of `φ`.
+-/
+
+namespace MulEquiv
+
+/-- A function `φ : G → G` is fixed-point-free if `1 : G` is the only fixed point of `φ`. -/
+def FixedPointFree {G : Type*} [One G] (φ : G → G) := ∀ g, φ g = g → g = 1
```

Comment:
I think this should not be in the `MulEquiv` namespace. But I'm not sure in which namespace it *should* be... `Function` is bad, because in that context the function really not have any fixed points. `MulHom`? `MonoidHom`?
Maybe we should ask on zulip.

Reply:
`MonoidHom` seems most natural to me (the `MulEquiv` was just a holdover from when I had to state several of the theorems in terms of `MulEquiv`)

Reply:
(`MonoidHom` seems better than `MulHom` since the whole point is that `1` is special).